pyproject.toml

```diff
@@ -7,12 +7,48 @@ requires-python = ">=3.13"
 classifiers = [
   "Programming Language :: Python :: 3 :: Only",
   "Programming Language :: Python :: 3.13",
+]
+dependencies = [
+  "beautifulsoup4>=4.12.3",
+  "fake-useragent>=1.5.1",
+  "imageio>=2.36.1",
+  "keras>=3.7",
+  "lxml>=5.3",
+  "matplotlib>=3.9.3",
+  "numpy>=2.1.3",
+  "opencv-python>=4.10.0.84",
+  "pandas>=2.2.3",
+  "pillow>=11",
+  "requests>=2.32.3",
+  "rich>=13.9.4",
+  "scikit-learn>=1.5.2",
+  "sphinx-pyproject>=0.3",
+  "statsmodels>=0.14.4",
+  "sympy>=1.13.3",
+  "tweepy>=4.14",
+  "typing-extensions>=4.12.2",
+  "xgboost>=2.1.3",
+]
 
+[dependency-groups]
+dev = [
+  "pytest>=8.3.4",
+  "pytest-cov>=6",
+]
+docs = [
+  "myst-parser>=4",
+  "sphinx-autoapi>=3.4",
+  "sphinx-pyproject>=0.3",
 ]
-optional-dependencies.docs = [
-  "myst-parser",
-  "sphinx-autoapi",
-  "sphinx-pyproject",
+euler-validate = [
+  "numpy>=2.1.3",
+  "pytest>=8.3.4",
+  "pytest-cov>=6",
+  "requests>=2.32.3",
+]
```

Comment:
Let's avoid adding `pytest` and `pytest-cov` here. We can specify the `--group` flag multiple times so `uv sync --group euler-validate --group dev`